### DISCUSSION TITLE
How to get only one component from dymamiczone using bookshelf

### DISCUSSION THREAD

**Discussion by ginishsharma** [strapi/strapi#6218] [Legacy Topics]
I have this dynamiczone field in the model and I need this field for some endpoints so I made this "autoPopulate": false now I need to fetch only one component for e.g "extra-options.file-upload" but it's not working

```
"extra_options": {
  "type": "dynamiczone",   
  "autoPopulate": false,   
  "components": [       
    "extra-options.file-upload",
    "extra-options.photo-gallery",       
    "extra-options.embed-link"
  ]
}
```

I'm using MySQL database and this is the query in which I'm trying to get the value

`entities = await strapi.query('model').find({ id: 1 }, ['extra-options.file-upload'])`

**Comment by lauriejim** [comment]
Hello! Due to the way Strapi manages DZ, you will not be able to chose the components you have in your response.

And right now, I don't have any workaround for this use case.
